pkg/manager/modelregistry.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manager
+
+import (
+	"fmt"
+	"github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/ygot/ygot"
+	log "k8s.io/klog"
+	"plugin"
+)
+
+// ModelPlugin is a set of methods that each model plugin should implement
+type ModelPlugin interface {
+	ModelData() (string, string, []*gnmi.ModelData)
+	UnmarshalConfigValues(jsonTree []byte) (*ygot.ValidatedGoStruct, error)
+	Validate(*ygot.ValidatedGoStruct, ...ygot.ValidationOption) error
+}
+
+// RegisterModelPlugin adds an external model plugin to the model registry at startup
+func (m *Manager) RegisterModelPlugin(moduleName string) error {
+	log.Info("Loading module", moduleName)
+	modelPluginModule, err := plugin.Open(moduleName)
```

Comment:
woudl be interesting to break this apart in order to have a REST api where a modelPlugin can be dropped in as a jar and added as information for a device.

Reply:
Can I create an issue to do this as a follow up? I didn't want to make this patch too big

Reply:
+1 also, not majorly important as of right now.